staging/src/k8s.io/apiserver/pkg/admission/metrics/metrics.go

```diff
@@ -217,19 +218,33 @@ func newAdmissionMetrics() *AdmissionMetrics {
 		},
 		[]string{"name", "type", "operation", "code", "rejected"})
 
+	webhookSLO := metrics.NewHistogramVec(
+		&metrics.HistogramOpts{
+			Namespace:      namespace,
+			Subsystem:      subsystem,
+			Name:           "webhook_slo_duration_seconds",
+			Help:           "Admission webhook latency histogram in seconds, identified by name and broken out for each operation and for a configurable set of API resources.",
+			Buckets:        []float64{0.005, 0.025, 0.1, 0.5, 1.0, 2.5, 10.0},
+			StabilityLevel: metrics.ALPHA,
+		},
+		[]string{"name", "operation", "group", "version", "resource", "subresource"},
```

Comment:
Do we really need the granularity provided by the `operation` label? Defining SLOs based on the actual operation made by the webhook seems far-fetched considering the complexity of setting it up. I would think that SLOs at the webhook type level (`validating`, `mutating`, `conversion`) are more likely to be used than at the operation level. But I am fairly new to that so I might be wrong here.

Reply:
I included `operation` with thinking that if a webhook could possibly have very different code paths for the different operations, and the performance of one operation might mask the bad performance of another. A quick survey of existing webhooks does not seem to support this (e.g. CREATE/UPDATE tend to do pretty much exactly the same thing ). I can be convinced to remove it.

As for webhook type, if we were to include `conversion`:
* We need a better name for the metric, since it is currently named `..._admission_webhook_admission_...` (just drop the `admission`s) ?
* We need to refactor the slo's metric observation code to be available to crd conversion.
* I think just `admission` vs `conversion` for webhook type might suffice. There is a possibility for name collision, but we already ignore the fact that two webhooks can have the same name if they exist in different configuration files, and would be extending that to include different types of configuration files.